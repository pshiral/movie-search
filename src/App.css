.app {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100vh;
}

header {
  color:dimgrey;
  margin: 1rem 0
}

.autocomplete {
  position: relative;
  width: 300px;
}

input {
  width: 100%;
  padding: 10px;
  font-size: 16px;
  box-sizing: border-box;
}

.options {
  position: absolute;
  border: 1px solid #d4d4d4;
  border-bottom: none;
  border-top: none;
  z-index: 2;
  top: 100%;
  left: 0;
  right: 0;
}

.options div {
  padding: 10px;
  cursor: pointer;
  background-color: #fff;
  border-bottom: 1px solid #d4d4d4;
}

.options div:hover {
  background-color: #e9e9e9;
}

.autocomplete-active {
  background-color: DodgerBlue !important;
  color: #ffffff;
}

.card {
  display: flex;
    flex-direction: column;
    border: 1px solid grey;
    padding: 20px;
    margin-top: 20px;
    width: 300px;
    box-sizing: border-box;
}

.title {
  display: flex;
  justify-content: space-between;
  padding: 10px 0;
}