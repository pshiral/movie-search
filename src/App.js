import { useCallback, useState } from 'react';
import './App.css';

function App() {
  const [options, setOptions] = useState([]),
    [selected, setSelected] = useState(null);

  const debounce = (cb) => {
    let id = null;

    return (...args) => {
      clearTimeout(id);

      id = setTimeout(() => {
        cb.apply(null, args);
      }, 300);
    };
  };

  const handleChange = useCallback(debounce((event) => {
    let text = event?.target?.value;

    if (text.length > 2) {
      fetch(`http://www.omdbapi.com/?s=${text}&apiKey=74de3445`).then((response) => {
        return response.json();
      }).then((response) => {
        const result = response?.Search?.slice(0, 5);

        if (result) {
          setOptions(response.Search.slice(0, 5));
        } else {
          setOptions([]);
        }
      });
    } else {
      setOptions([]);
    }
  }), []);

  return (
    <div className="app">
      <header>
        <h2 className="heading">
          Web Developer Assignment
        </h2>
        <span className="by">Sr Software Engineer</span>
      </header>
      <main>
        <form autoComplete="off">
          <section className="autocomplete">
            <input type="text" name="places" placeholder="Search a place"
              data-test-id="input" onChange={handleChange} />
            <div className="options">
              {options?.map((movie) => (
                <div onClick={() => {
                  setOptions([]);
                  setSelected(movie);
                }}>{movie.Title}</div>
              ))}
            </div>
          </section>
          {selected && <section className="card">
            <img alt={selected.Title} src={selected.Poster} />
            <div className="title">
              <span>Title: {selected.Title}</span>
              <span>Year: {selected.Year}</span>
            </div>
          </section>}
        </form>
      </main>
    </div>
  );
}

export default App;
