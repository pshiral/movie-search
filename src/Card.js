import './App.css';

function Card({ selected }) {
    return (
        <section className="card">
            <img alt={selected.Title} src={selected.Poster} />
            <div className="title">
                <span>Title: {selected.Title}</span>
                <span>Year: {selected.Year}</span>
            </div>
        </section>
    );
}

export default Card;
