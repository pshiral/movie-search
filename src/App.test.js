import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import Card from './Card';

describe('App', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(<App />));

  it('should render correctly', () => expect(wrapper).toMatchSnapshot());

  it('should have a root element', () => {
    expect(wrapper.find('div.app').length).toEqual(1);
  });

  it('should render card correctly', () => {
    const selected = { Title: 'My test', Year: '2021', Poster: 'test' };

    const component = shallow(<Card selected={selected} />);
    expect(component).toMatchSnapshot();
  });
});