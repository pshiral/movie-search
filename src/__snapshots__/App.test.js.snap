// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render card correctly 1`] = `ShallowWrapper {}`;

exports[`App should render correctly 1`] = `ShallowWrapper {}`;
